server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala

```diff
@@ -361,8 +360,16 @@ private[deploy] class Worker(
   ShutdownHookManager.get().addShutdownHook(
     new Thread(new Runnable {
       override def run(): Unit = {
+        logInfo("Shutdown hook called.")
         shutdown.set(true)
-        // TODO: call stop after all reserved slot finished commit/destroy
+        val interval = RssConf.checkSlotsFinishedInterval(conf)
+        val timeout = RssConf.checkSlotsFinishedTimeoutMs(conf)
+        var waitTimes = 0
+        while (!partitionLocationInfo.isEmpty && waitTimes * interval < timeout) {
```

Comment:
Will you add some log to indicate that this block is waiting too long?

Reply:
How about current?